app/uk/gov/hmrc/agentmappingfrontend/auth/AuthActions.scala

```diff
@@ -99,4 +100,23 @@ trait AuthActions extends AuthorisedFunctions with AuthRedirects {
         case _: AuthorisationException => toGGLogin(s"${appConfig.authenticationLoginCallbackUrl}${request.uri}")
       }
 
+  def withCheckForArn(body: Option[EnrolmentIdentifier] => Future[Result])(
+    implicit request: Request[AnyContent],
+    hc: HeaderCarrier,
+    ec: ExecutionContext): Future[Result] =
+    authorised(AuthProviders(GovernmentGateway) and AffinityGroup.Agent)
+      .retrieve(allEnrolments) {
+        case agentEnrolments =>
+          val optHmrcEnrolIden = agentEnrolments.getEnrolment("HMRC-AS-AGENT") match {
```

Comment:
You can use `agentEnrolments.getEnrolment("HMRC-AS-AGENT").map(_.asAgentEnrol.getIdentifier("AgentReferenceNumber"))` instead of pattern matching